### PR TITLE
Add Docker-compose setting for IT testing

### PR DESCRIPTION
**Pull request by salyh**
### Description
Add Docker-compose setting for IT testing

### Issues Resolved
https://github.com/opensearch-project/opensearch-spark/issues/601

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by salyh** [comment]
I need still to wire spark and opensearch and add some docs ...

**Comment by YANG-DB** [comment]
nice
can you also add opensearch dashbords to the docker compose so that we can acutally test ppl directly from the query workbench ?
thanks ...
see example [here](https://github.com/opensearch-project/opentelemetry-demo/blob/4108ea8918dac81a97234a9181b01d4334429def/docker-compose.yml#L827)

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      nofile:
+        soft: 65536
+        hard: 65536
+    volumes:
+      - ppl-opensearch-data:/usr/share/opensearch/data
+    ports:
+      - 9200:9200
+    networks:
+      - ppl-opensearch-net
+  ppl-opensearch-dashboards:
```
Comment (salyh) [inline docker-compose.yml:26]: OSD

**Comment by salyh** [comment]
Opensearch Dashboards are already there, see https://github.com/opensearch-project/opensearch-spark/pull/606/files#r1736775479

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+---
+services:
+  ppl-opensearch-node:
```
Comment (dai-chen) [inline docker-compose.yml:3]: can we make this generic Docker setup instead of only PPL?

Comment (YANG-DB) [reply docker-compose.yml:3]: thats is a good idea

**Review by YANG-DB — COMMENTED** [review]
please add a section in the readme file explaining how to run this and test PPL/SQL queries against this spark service...

**Comment by salyh** [comment]
@YANG-DB any update?

File: docker-compose/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,4 @@
+FROM bitnami/spark:3.5
```
[the remark this answers is not in the document]
Comment (normanj-bitquill) [reply docker-compose/Dockerfile:1]: I started some work on running with the EMR Spark image.

https://github.com/opensearch-project/opensearch-spark/issues/647#issuecomment-2508715293

If there is interest, I could put up a PR. I would keep what you have here, but add a separate Dockerfile for EMR Spark.

**Comment by YANG-DB** [comment]
closing this as its being addressed [here](https://github.com/opensearch-project/opensearch-spark/pull/965)

**Pull request closed without merging**